libmcount/dynamic.c

```diff
@@ -127,15 +129,25 @@ static int do_dynamic_update(struct symtabs *symtabs, char *patch_funcs)
 				stats.skipped++;
 				break;
 			case 0:
+				success = true;
 			default:
 				break;
 			}
 			stats.total++;
 		}
 
+		if (!success) {
+			nr_failed++;
+			func_name = name;
+		}
+
 		name = strtok(NULL, ";");
 	}
 
+	if (nr_failed)
+		pr_out("%s cannot be patched dynamically\n",
+		       nr_failed > 1 ? "some functions" : func_name);
+
```

Comment:
Looks good But we can miss a case the user gave **more than just one wrong function names** with `-P`.
```
$ uftrace -v -P wrong_name1 -P wrong_name2 ./hello.dyn
```
In that case, `stats.total`, `stats.failed` and `stats.skipped` are 0 zero value like below. 
```
$ uftrace -v -P wrong_name1 -P wrong_name2 ./hello.dyn
uftrace: creating 1 thread(s) for recording
uftrace: using libmcount.so library for tracing
mcount: initializing mcount library
dynamic: dynamic update stats:
dynamic:    total:        0
dynamic:  patched:        0 (0.00%)
dynamic:   failed:        0 (0.00%)
dynamic:  skipped:        0 (0.00%)
mcount: mcount setup done
mcount: shmem_finish: tid: 25703 seqnum = 0 curr = 0, nr_buf = 2 max_buf = 2
hello
foo
bar
uftrace: child terminated with exit code: 4
uftrace: reading /tmp/uftrace-live-TllTZr/task.txt file
uftrace: live-record finished.. 
uftrace: start live-replaying...
uftrace: reading /tmp/uftrace-live-TllTZr/task.txt file
# DURATION    TID     FUNCTION
   0.696 us [25703] | __monstartup();
   0.493 us [25703] | __cxa_atexit();
   0.837 us [25703] | puts();
   0.218 us [25703] | puts();
   0.127 us [25703] | puts();
```